ui/client/components/ToolBox.tsx

```diff
@@ -39,8 +39,7 @@ export default function ToolBox() {
     <div id="toolbox">
       <div>
         {nodesToAdd.map(nodeGroup => {
-          const label =
-            <span className={"group-label"} onClick={() => toggleGroup(nodeGroup)}>{nodeGroup.name}</span>
+          const label = <span className={"group-label"} onClick={() => toggleGroup(nodeGroup)}>{nodeGroup.name}</span>
```

Comment:
Now you create for feach nodesToAdd new function. It's not good. Please create somethig like this:

```
clickGroupHandler = (node) => (node) => {

}
```
And use it:
```
 onClick={toggleGroup(clickGroupHandler)}
```
More info you can find here: https://medium.com/@machnicki/handle-events-in-react-with-arrow-functions-ede88184bbb

Reply:
it's not in scope of my change.